### ISSUE TITLE
Clone logger with different channel name

### ISSUE THREAD

**Issue by adambro** [Seldaek/monolog#730]
I'd like to add another logger which is different channel (different component in application). It seems to be a case described in documentation:
https://github.com/Seldaek/monolog/blob/master/doc/01-usage.md#leveraging-channels

The problem is in order to avoid sharing handlers and processors in Dependency Injection Container I had to write ugly piece of code as below:. The main point here is to keep the same UID processor instance for both loggers:
```php
    $logger = new \Monolog\Logger('elasticsearch');
    // reuse UID processor and handlers
    $logger->setHandlers($c['logger']->getHandlers());
    foreach ($c['logger']->getProcessors() as $proc) {
        $logger->pushProcessor($proc);
    }
```

Easier (and more performant) would be to just clone the logger object and change it's channel name, like:
```php
$logger2 = clone $logger;
$logger2->setName('elasticsearch');
```

**Comment by Seldaek** [comment]
Fixed by cb2778ae77fa64003afb1967e30a4aea07f4503b - you can do `$logger2 = $logger->withName('elasticsearch');`

**Comment by adambro** [comment]
Thank you @Seldaek
